src/data_model/transformations.jl

```diff
@@ -472,3 +472,917 @@ function apply_voltage_angle_difference_bounds!(eng::Dict{String,<:Any}, vad::Re
+The delta multiplier controls the scaling of bounds of delta-connected units.
+"""
+function add_unit_vbounds!(data_eng::Dict{String,Any}; lb_pu::Real=0.9, ub_pu::Real=1.1, delta_multiplier::Real=sqrt(3), unit_comp_types::Vector{<:AbstractString}=["load", "generator", "storage", "pv"])
+    bus_vbase, line_vbase = calc_voltage_bases(data_eng, data_eng["settings"]["vbases_default"])
+    for comp_type in intersect(unit_comp_types, keys(data_eng))
+        for (id, unit) in data_eng[comp_type]
+            @assert _is_oneport_component(unit) "The $comp_type $id is not a oneport component."
+            bus_id = unit["bus"]
+            bus = data_eng["bus"][bus_id]
+            unit_vm_pair_lb, unit_vm_pair_ub = _generate_vm_pairs(unit["connections"], unit["configuration"], lb_pu*bus_vbase[bus_id], ub_pu*bus_vbase[bus_id], delta_multiplier=delta_multiplier)
+            bus_vm_pair_lb = haskey(bus, "vm_pair_lb") ? bus["vm_pair_lb"] : []
+            bus_vm_pair_ub = haskey(bus, "vm_pair_ub") ? bus["vm_pair_ub"] : []
+            bus["vm_pair_lb"] = unique(vcat(bus_vm_pair_lb, unit_vm_pair_lb))
+            bus["vm_pair_ub"] = unique(vcat(bus_vm_pair_ub, unit_vm_pair_ub))
+        end
+    end
+end
+
+
+"""
+Adds symmetric phase-to-neutral and phase-to-phase voltage bounds when possible
+for each bus through the three-phase bus syntax.
+"""
+function add_bus_pn_pp_ng_vbounds!(data_eng::Dict{String,Any}, phase_terminals::Vector, neutral_terminal;
+    pn_lb_pu::Union{Real,Missing}=missing, 
+    pn_ub_pu::Union{Real,Missing}=missing, 
+    pp_lb_pu::Union{Real,Missing}=missing, 
+    pp_ub_pu::Union{Real,Missing}=missing, 
+    ng_ub_pu::Union{Real,Missing}=missing,  
+)
+    bus_vbase, _ = calc_voltage_bases(data_eng, data_eng["settings"]["vbases_default"])
+
+    for (id,bus) in data_eng["bus"]
+        vbase = bus_vbase[id]
+        terminals = bus["terminals"]
+
+        has_neutral = neutral_terminal in terminals
+        if has_neutral
+            bus["neutral"] = neutral_terminal
+        end
+        bus_phases = intersect(phase_terminals, terminals)
+        has_phases = !isempty(bus_phases)
+        if has_phases
+            bus["phases"] = bus_phases
+        end
+
+        if has_neutral && !ismissing(ng_ub_pu)
+            bus["vm_ng_ub"] = ng_ub_pu*vbase
+        end
+        if has_phases && has_neutral && !ismissing(pn_lb_pu)
+            bus["vm_pn_lb"] = pn_lb_pu*vbase
+        end
+        if has_phases && has_neutral && !ismissing(pn_ub_pu)
+            bus["vm_pn_ub"] = pn_ub_pu*vbase
+        end
+        if has_phases && length(bus_phases)>1 && !ismissing(pp_lb_pu)
+            bus["vm_pp_lb"] = pp_lb_pu*vbase
+        end
+        if has_phases && length(bus_phases)>1 && !ismissing(pp_ub_pu)
+            bus["vm_pp_ub"] = pp_ub_pu*vbase
+        end
+    end
+end
+
+
+"Calculate no-load starting values for all bus-terminals pairs."
```

Comment:
For exported functions, we should follow the new docstring pattern where the function is given (indented) first, see, e.g. transform_data_model function as an example.